frontend/server/src/DAO/Courses.php

```diff
@@ -160,59 +163,74 @@ public static function getStudentsInCourseWithProgressPerAssignment(
                         FROM Assignments a
                         INNER JOIN Problemsets ps
                             ON a.problemset_id = ps.problemset_id
                         INNER JOIN Problemset_Problems psp
                             ON psp.problemset_id = ps.problemset_id
+                        INNER JOIN Problems p
+                            ON p.problem_id = psp.problem_id
                         INNER JOIN Submissions s
-                            ON s.problem_id = psp.problem_id
+                            ON s.problem_id = p.problem_id
                             AND s.problemset_id = a.problemset_id
                         INNER JOIN Runs r
                             ON r.run_id = s.current_run_id
                         WHERE a.course_id = ?
-                        GROUP BY a.assignment_id, psp.problem_id, s.identity_id
+                        GROUP BY a.assignment_id, p.problem_id, s.identity_id
+                        ORDER BY p.alias
                     ) bpr
-                    GROUP BY bpr.assignment_id, bpr.identity_id
+                    GROUP BY bpr.assignment_id, bpr.problem_id, bpr.identity_id
                 ) pr
                 ON pr.identity_id = i.identity_id';
 
-        /** @var list<array{assignment_alias: null|string, assignment_score: float|null, name: null|string, username: string}> */
+        /** @var list<array{assignment_alias: null|string, name: null|string, problem_alias: null|string, problem_score: float|null, username: string}> */
         $rs = \OmegaUp\MySQLConnection::getInstance()->GetAll(
             $sql,
             [$groupId, $courseId]
         );
-        $progress = [];
+
+        $allProgress = [];
         foreach ($rs as $row) {
             $username = $row['username'];
-            if (!isset($progress[$username])) {
-                $progress[$username] = [
+            if (!isset($allProgress[$username])) {
+                $allProgress[$username] = [
                     'name' => $row['name'],
                     'progress' => [],
                     'username' => $username,
                 ];
             }
 
-            if (
-                !is_null($row['assignment_score']) &&
-                !is_null($row['assignment_alias'])
-            ) {
-                $progress[$username]['progress'][$row['assignment_alias']] = $row['assignment_score'];
+            $assignmentAlias = $row['assignment_alias'];
+            $problemAlias = $row['problem_alias'];
+
+            if (is_null($assignmentAlias) || is_null($problemAlias)) {
+                continue;
+            }
+
+            if (!isset($allProgress[$username]['progress'][$assignmentAlias])) {
+                $allProgress[$username]['progress'][$assignmentAlias] = [];
             }
+
+            $allProgress[$username]['progress'][$assignmentAlias][$problemAlias] = is_null(
+                $row['problem_score']
+            ) ? 0.0 : $row['problem_score'];
```

Comment:
nit:

```suggestion
            $allProgress[$username]['progress'][$assignmentAlias][$problemAlias] = flotaval(
                $row['problem_score']
            );
```

obtains the same result.